app/controllers/CheckYourAnswersController.scala

```diff
@@ -77,18 +79,32 @@ class CheckYourAnswersController @Inject()(
         ).flatten
       ).withCssClass("govuk-!-margin-bottom-9")
 
-      val totalVatToCountries = service.getVatOwedToEuCountries(request.userAnswers)
-      val totalVatOnSales = service.getTotalVatOnSales(request.userAnswers)
+      val containsCorrections = request.userAnswers.get(AllCorrectionPeriodsQuery).isDefined
+
+      val totalVatToCountries = service.getVatOwedToEuCountries(request.userAnswers).filter(vat => vat.totalVat > 0)
+      val noPaymentDueCountries = service.getVatOwedToEuCountries(request.userAnswers).filter(vat => vat.totalVat <= 0)
+      val totalVatOnSales = service.getTotalVatOwedAfterCorrections(request.userAnswers)
```

Comment:
would be good if we could have a look at the failing test and add in something to tell the mock for this service to return what we expect. As there is logic associated with the value returned from here to conditionally show text. Its different to what they were before hand as they defaulted to zero so I dont think they had logic associated with them

Reply:
we didn't need to mock the service in the tests for this as we are using appropriate user answers for each one, so we can use the actual service